core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -449,21 +605,32 @@ private[spark] class MapOutputTrackerMaster(
       try {
         while (true) {
           try {
-            val data = mapOutputRequests.take()
-             if (data == PoisonPill) {
+            val data = mapOutputTrackerMasterMessages.take()
+            if (data == PoisonPill) {
               // Put PoisonPill back so that other MessageLoops can see it.
-              mapOutputRequests.offer(PoisonPill)
+              mapOutputTrackerMasterMessages.offer(PoisonPill)
               return
             }
-            val context = data.context
-            val shuffleId = data.shuffleId
-            val hostPort = context.senderAddress.hostPort
-            logDebug("Handling request to send map output locations for shuffle " + shuffleId +
-              " to " + hostPort)
-            val shuffleStatus = shuffleStatuses.get(shuffleId).head
-            context.reply(
-              shuffleStatus.serializedMapStatus(broadcastManager, isLocal, minSizeForBroadcast,
-                conf))
+
+            data match {
+              case GetMapStatusMessage(shuffleId, context) =>
+                val hostPort = context.senderAddress.hostPort
+                val shuffleStatus = shuffleStatuses.get(shuffleId).head
+                logDebug("Handling request to send map output locations for shuffle " + shuffleId +
+                  " to " + hostPort)
+                context.reply(
+                  shuffleStatus.serializedOutputStatus(broadcastManager, isLocal,
+                    minSizeForBroadcast, conf, isMapOutput = true))
+
+              case GetMergeStatusMessage(shuffleId, context) =>
+                val hostPort = context.senderAddress.hostPort
+                val shuffleStatus = shuffleStatuses.get(shuffleId).head
+                logDebug("Handling request to send merge output locations for" +
+                  " shuffle " + shuffleId + " to " + hostPort)
+                context.reply(
+                  shuffleStatus.serializedOutputStatus(broadcastManager, isLocal,
+                    minSizeForBroadcast, conf, isMapOutput = false))
+            }
```

Comment:
Both `GetMapStatusMessage` and `GetMergeStatusMessage` are doing pretty much same thing except for log message difference and `isMapOutput` flag difference.
Add a local method and delegate to it to minimize duplication.

Reply:
As part of we would be adding one more message for getting shuffle mergers https://issues.apache.org/jira/browse/SPARK-34826 which is why made this change.

Reply:
That would be for a new message right ? Will it have changes to these existing messages ? If not, would something like this not work ?

```

def handleStatusMessage(shuffleId: Int, context: RpcCallContext, messageType: String, mapOutput: Boolean): Unit = {
  val hostPort = context.senderAddress.hostPort
  val shuffleStatus = shuffleStatuses.get(shuffleId).head
  logDebug(s"Handling request to send $messageType output locations for shuffle $shuffleId to $hostPort")
  context.reply(
    shuffleStatus.serializedOutputStatus(broadcastManager, isLocal, minSizeForBroadcast, conf, isMapOutput = mapOutput))
}

data match {
  case GetMapStatusMessage(shuffleId, context) => handleStatusMessage(shuffleId, context, "map", true)
  case GetMergeStatusMessage(shuffleId, context) => handleStatusMessage(shuffleId, context, "map", false)
}

```

Reply:
Addressed this comment.

Reply:
Just realized I totally misunderstood the first comment above @mridulm